platforms/core-runtime/io/src/main/java/org/gradle/internal/io/StreamByteBuffer.java

```diff
@@ -41,16 +42,17 @@
  * Reading the buffer will clear the buffer.
  * This is not thread-safe, it is intended to be used by a single Thread.
  */
+@SuppressWarnings("NullAway")
```

Comment:
❔  Why are we suppressing the whole class and not individual methods? What are we suppressing?

Reply:
> ❔ Why are we suppressing the whole class and not individual methods? What are we suppressing?

It is very convoluted with regards to nullability guarantees. Rather than tackling it right now, I decided to postpone it to unblock other projects. I'll add a TODO comment.